app/src/main/java/de/tum/in/tumcampusapp/managers/ChatRoomManager.java

```diff
@@ -58,52 +52,59 @@ public ChatRoomManager(Context context) {
      * @param joined chat room 1=joined, 0=not joined/left chat room, -1=not joined
      * @return List of chat messages
      */
-    public Cursor getAllByStatus(int joined) {
-        String condition = "joined=1";
+
+
+    public List<ChatRoomDbRow> getAllByStatus(int joined) {
+        List<ChatMessage> msgs;
+        List<ChatRoomDbRow> rooms;
         if (joined == 0) {
-            condition = "joined=0 OR joined=-1";
+            msgs = chatRoomDao.getTsAndTextRoomsNotJoined();
+            rooms = chatRoomDao.getAllRoomsNotJoinedList();
+            for(int i=0;i<rooms.size();i++) {
+                if(msgs.get(i).getTimestamp()!=null && msgs.get(i).getText()!=null) {
+                    rooms.get(i).setTimestamp(msgs.get(i).getTimestamp());
+                    rooms.get(i).setText(msgs.get(i).getText());
+                }
+            }
+            return rooms;
         }
+        msgs = chatRoomDao.getTsAndTextRoomsJoined();
+        rooms = chatRoomDao.getAllRoomsJoinedList();
+        for(int i=0;i<rooms.size();i++) {
+            if (msgs.get(i).getTimestamp()!=null && msgs.get(i).getText()!=null)    {
+                rooms.get(i).setTimestamp(msgs.get(i).getTimestamp());
+                rooms.get(i).setText(msgs.get(i).getText());
+            }
 
-        return db.rawQuery("SELECT r.*, m.ts, m.text " +
-                           "FROM chat_room r " +
-                           "LEFT JOIN (SELECT MAX(timestamp) ts, text, room FROM chat_message GROUP BY room) m ON (m.room=r.room) " +
-                           "WHERE " + condition + " " +
-                           " " +
-                           "ORDER BY r.semester!='', r.semester_id DESC, datetime(m.ts) DESC, r.name", null);
+        }
+        return rooms;
```

Comment:
this method seems like doing this join manually...
why do we need 20 lines of java here instead of that SQL join?

Reply:
The problem was that sometimes the timestamp and text returned by the Queries were null and then I got an error for trying to write null into a non-null attribute. But now I figured out how to make an attribute able to be null as well. So the we don*t need the manual join anymore.